pkgs/development/compilers/dmd/2.067.1.nix

```diff
@@ -1,47 +1,135 @@
+    repo = "druntime";
+    rev = "v${version}";
+    sha256 = "1n2qfw9kmnql0fk2nxikispqs7vh85nhvyyr00fk227n9lgnqf02";
+  })
+  (fetchFromGitHub {
+    owner = "dlang";
+    repo = "phobos";
+    rev = "v${version}";
+    sha256 = "0fywgds9xvjcgnqxmpwr67p3wi2m535619pvj159cgwv5y0nr3p1";
+  })
+  ];
+
+  sourceRoot = ".";
+
+  postUnpack = ''
+      mv dmd-v${version}-src dmd
+      mv druntime-v${version}-src druntime
+      mv phobos-v${version}-src phobos
+  '';
+
+  # Compile with PIC to prevent colliding modules with binutils 2.28.
+  # https://issues.dlang.org/show_bug.cgi?id=17375
+  usePIC = "-fPIC";
+
+  postPatch = ''
+      # Ugly hack so the dlopen call has a chance to succeed.
+      # https://issues.dlang.org/show_bug.cgi?id=15391
+      substituteInPlace phobos/std/net/curl.d \
+          --replace libcurl.so ${curl.out}/lib/libcurl.so
 
-  nativeBuildInputs = [ unzip makeWrapper ];
+      # Ugly hack to fix the hardcoded path to zoneinfo in the source file.
+      # https://issues.dlang.org/show_bug.cgi?id=15391
+      substituteInPlace phobos/std/datetime.d \
+          --replace /usr/share/zoneinfo/ ${tzdata}/share/zoneinfo/
 
-  postPatch = stdenv.lib.optionalString stdenv.isDarwin ''
-      # Allow to use "clang++", commented in Makefile
-      substituteInPlace src/dmd/posix.mak \
-          --replace g++ clang++ \
-          --replace MACOSX_DEPLOYMENT_TARGET MACOSX_DEPLOYMENT_TARGET_
+      substituteInPlace druntime/test/shared/Makefile \
+          --replace "DFLAGS:=" "DFLAGS:=${usePIC} "
 
-      # Was not able to compile on darwin due to "__inline_isnanl"
-      # being undefined.
-      substituteInPlace src/dmd/root/port.c --replace __inline_isnanl __inline_isnan
+      # phobos uses curl, so we need to patch the path to the lib.
+      substituteInPlace phobos/posix.mak \
+          --replace "-soname=libcurl.so.4" "-soname=${curl.out}/lib/libcurl.so.4"
+
+      # Use proper C++ compiler
+      substituteInPlace dmd/src/posix.mak \
+          --replace g++ $CXX
   ''
+
     + stdenv.lib.optionalString stdenv.isLinux ''
-        substituteInPlace src/dmd/root/port.c \
+        substituteInPlace dmd/src/root/port.c \
           --replace "#include <bits/mathdef.h>" "#include <complex.h>"
-      '';
+    ''
+
+    + stdenv.lib.optionalString stdenv.isDarwin ''
+        substituteInPlace dmd/src/posix.mak \
+            --replace MACOSX_DEPLOYMENT_TARGET MACOSX_DEPLOYMENT_TARGET_
+
+        # Was not able to compile on darwin due to "__inline_isnanl"
+        # being undefined.
+        substituteInPlace dmd/src/root/port.c --replace __inline_isnanl __inline_isnan
+    '';
+
+  nativeBuildInputs = [ makeWrapper unzip which ];
+  buildInputs = [ curl tzdata ];
 
   # Buid and install are based on http://wiki.dlang.org/Building_DMD
   buildPhase = ''
-      cd src/dmd
+      cd dmd
       make -f posix.mak INSTALL_DIR=$out
-      export DMD=$PWD/dmd
+      ${
+          let bits = if stdenv.is64bit then "64" else "32";
```

Comment:
There's a bunch of things in that attrset. Try `stdenv.hostPlatform.parsed.cpu.bits` maybe?